microservices/external_data_query/external_data_query/rcsb_pdb/services.py

```diff
@@ -0,0 +1,209 @@
+    search_url = "https://search.rcsb.org/rcsbsearch/v2/query"
+    headers = {'Content-Type': 'application/json'}
+    query_payload = {
+        "query": {
+            "type": "group",
+            "logical_operator": "and",
+            "nodes": [
+                {
+                    "type": "terminal",
+                    "service": "text",
+                    "parameters": {
+                        "attribute": "rcsb_entry_info.selected_polymer_entity_types",
+                        "operator": "exact_match",
+                        "value": "Protein (only)"
+                    }
+                },
+                {
+                    "type": "terminal",
+                    "service": "text",
+                    "parameters": {
+                        "attribute": "struct.title",
+                        "operator": operator,
+                        "value": protein_name
+                    }
+                }
+            ]
+        },
+        "request_options": request_options,
+        "return_type": "entry"
+    }
+
+    print(query_payload)
+    search_response = requests.post(search_url, headers=headers, data=json.dumps(query_payload))
+    fetched_proteins = []
+
+    fasta_base_url = "https://www.rcsb.org/fasta/entry/"
+    pdb_base_link = "https://www.rcsb.org/structure/"
+
+    if search_response.status_code == 200:
+        search_results = search_response.json()
+
+        for result in search_results.get("result_set", []):
+            pdb_id = result.get("identifier")
+            fasta_response = requests.get(f"{fasta_base_url}{pdb_id}")
+
+            if fasta_response.status_code == 200:
+                fasta_contents = fasta_response.text
+                link = f"{pdb_base_link}{pdb_id}"
+                fetched_proteins.append(FetchedProtein(fasta_contents=fasta_contents, link=link))
+            else:
+                print(f"Failed to fetch FASTA for PDB ID: {pdb_id}")
+
+        return GetFastaFilesResponse(fasta_contents=fetched_proteins)
+
+    return GetFastaFilesResponse(fasta_contents=[])
+
+
+def fetch_entries_by_complex_query(request: ComplexQueryRequest) -> GetFastaFilesResponse:
+    search_url = "https://search.rcsb.org/rcsbsearch/v2/query"
+    headers = {'Content-Type': 'application/json'}
+    fasta_base_url = "https://www.rcsb.org/fasta/entry/"
+    pdb_base_link = "https://www.rcsb.org/structure/"
+
+    def build_query(node: QueryNode):
+        if isinstance(node, TerminalNode):
+            parameters = {
+                "attribute": node.attribute,
+                "operator": node.operator,
+                "value": node.value
+            }
+            # Only add 'negation' if it is True
+            if node.negation:
+                parameters["negation"] = node.negation
+            return {
+                "type": "terminal",
+                "service": "text",
+                "parameters": parameters
+            }
+        elif isinstance(node, LogicalNode):
```

Comment:
if node is not TerminalNode and not LogicalNode then this function will return null, consider using throw ValueError or removing elif

Reply:
Error is there now